app/routes.py

```diff
@@ -1 +1,227 @@
-from flask import Blueprint
\ No newline at end of file
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response, request, abort
+from sqlalchemy import asc, desc 
+import datetime
+import requests
+import os 
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["POST"])
+def create_tasks():
+    request_body = request.get_json()
+    new_task = Task.from_dict(request_body) 
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    task_dict = {}
+    task_dict["task"] = {"id":new_task.task_id, "title": new_task.title,"description":new_task.description, "is_complete":bool(new_task.completed_at)}
+
+    return make_response(jsonify(task_dict), 201) 
+
+@tasks_bp.route("", methods=["GET"])
+def get_all_tasks():
+    sort_query = request.args.get("sort")
+    if sort_query:
+        if "asc" in sort_query: 
+            tasks = Task.query.order_by(Task.title.asc())
+        elif "desc" in sort_query: 
+            tasks = Task.query.order_by(Task.title.desc()) 
+    else: 
+        tasks = Task.query.all()
+
+    tasks_response = [task.to_dict() for task in tasks]
+    return jsonify(tasks_response) 
+    
+
+def validate_model(cls, model_id):
+    try: 
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message": f"{cls.__name__} {model_id} invalid"}, 400)) 
+
+    model = cls.query.get(model_id)
+    
+    if not model:
+        abort(make_response({"message": f"{cls.__name__} {model_id} not found"}, 404)) 
+
+    return model
+
+@tasks_bp.route("/<task_id>", methods = ["GET"])
+def get_one_task(task_id):
+    task = validate_model(Task,task_id)
+
+    task_dict = {}
+    task_dict["task"] = {"id":task.task_id, "title": task.title,"description":task.description, "is_complete":bool(task.completed_at)}
+
+    if bool(task.goal_id) == True:
```

Comment:
Minor thing: this `if` condition can be simplified. First since `if` and all other conditional blocks already expect a `bool` that is either `True` or `False`, you could drop the `== True` and just write `if bool(task.goal_id):`.

But even more than that, Python already sort of calls `bool` for you in these conditional statements as that's kind of what checking the "truthiness" of the value means. With that, we can get down to `if task.goal_id:` as I used in my suggestion, and is the Pythonic way to write this.

Not that there aren't cases where your way of writing it may be useful, particularly if you wanted to perhaps treat a particular value different. Like perhaps `None` should mean something different than `False` even though both are falsy.